docs/http-api/projects.rst

```diff
@@ -0,0 +1,107 @@
+Projects
+========
+
+Get list of projects
+--------------------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  GET /projects
+
+Response
+^^^^^^^^
+
+* `200`:
+
+.. code-block:: json
+
+  [
+    {
+      "name": "first project",
+      "id": 1,
+      "author": "user_name",
+      "creation_date": "1970-01-01 00:00:00.000000 "
+    }
+  ]
+
+Get project
+-----------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  GET /projects/1
+
+Response
+^^^^^^^^
+
+* `200`:
+
+.. code-block:: json
+
+  {
+    "name": "first project",
+    "id": 1,
+    "author": "user_name",
+    "creation_date": "1970-01-01 00:00:00.000000 "
+  }
+
+* `404`: if given project doesn't exist
+
+
+Create project
+--------------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  POST /projects
+
+.. code-block:: json
+
+  {
+    "name": "first project"
+  }
+
+Response
+^^^^^^^^^^^^^^
+
+* `201`:
+
+.. code-block:: json
+
+  {
+    "name": "first project",
+    "id": 1,
+    "author": "user_name",
+    "creation_date": "1970-01-01 00:00:00.000000 "
+  }
+
+* `400`: if project with given name already exists
+
+
+Delete project
+--------------
+
+Request
+^^^^^^^
+
+.. code-block::
+
+  DELETE /projects/1?cascade=1
+
+* `cascade`: (optional, default - `0`) delete cascadely (with referenced tasks, models, etc)
+
+Response
+^^^^^^^^^^^^^^
+
+* `204`: OK
+* `400`: if `cascade` is not `1` and given project has tasks in it
+* `404`: if given project doesn't exist
```

Comment:
what about update?

Reply:
This is really a good question. `Project` has `id`, `name`, `author` and `creation_date`. `Task` has the same attributes and `project_id`. I believe the only thing which could be safely updated here is `name`. But do we actually need it? I don't know.

Reply:
Potentially there will be more fields. For example, description?
In task it is almost certain - when we add datasets and metrics.
Also, if you look at it from admin UI perspective - yes, we may need a way to change all the fields (for example, move tasks between projects, changing projects author)

Reply:
fixed